### PR TITLE
[vector_store]: Rename `SupabaseVector_` to `PgVector_` to avoid confusion

### LINKED ISSUE #6058: Using llama index with existing postgres db and pgvector extension
### Question Validation

- [X] I have searched both the documentation and discord for an answer.

### Question

Hi,

thanks for the great work ! I am a bit lost when using llama index for paragraph retrieval together with an existing postgres db and a table that already has embeddings computed. 

I am initiating like this:

`vector_store = SupabaseVectorStore(
    postgres_connection_string="postgresql://user:password@host:5432/defaultdb", 
    collection_name='news_paragraphs',
    dimension=768
)`

I have 2 questions:

1.  How can I specify the table to use and the field that contains the embeddings ? Is it collection_name ?
2. Once I did that how can I query the vector_store ? 

Really appreciate any help

### PR DESCRIPTION
**Pull request by jon-chuang**
# Description
Existing code works with pgvector extension on any postgres database. Make this explicit and improve info in docs.

Fixes #6058 #499 

## Type of Change

- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update (done)

# How Has This Been Tested?
- [X] Example in docs works locally
- [ ] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: llama_index/vector_stores/faiss.py [hunk 1]
```diff
@@ -6,20 +6,16 @@
 
 import logging
 import os
-import fsspec
```
Comment (jon-chuang) [inline llama_index/vector_stores/faiss.py:9]: My vscode formatter is a bit aggressive. Should I turn this off or is it fine?

Comment (jon-chuang) [reply llama_index/vector_stores/faiss.py:9]: Oops, just looked at contributing.md

**Review by jerryjliu — COMMENTED** [review]
@jon-chuang this makes sense. for BW compatibility purposes, would you still be able to maintain a SupabaseVectorStore class? Can define `SupabaseVectorStore = PgVectorStore` in `llama_index/vector_stores/__init__.py`, and explicitly make a comment saying that this is for backwards compat

**Comment by jon-chuang** [comment]
I also decided to make a superclass with a deprecation warning on init.

I've decided not to leave supabase in the regsitry. The registry is currently not used in any examples or mentioned in docs. Please correct me if this is mistaken.

**Comment by jon-chuang** [comment]
Oops, there is a duplicate now: https://github.com/jerryjliu/llama_index/pull/6190

I think we should merge them and use either vecs or psycopg.

I'm more keen on psycopg as vecs is just a thin wrapper, and more of a mainstream dependency.

**Comment by jon-chuang** [comment]
Closing this for now. I think both pg and supabase can coexist, they differ slightly in deps. Perhaps we can clean up and unify if users report that it is confusing

**Pull request closed without merging**